ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -162,10 +162,13 @@ azure_connection::read_object(
     return 0;
 }
 
+// Check if an object exists in the bucket and find the object size.
 int
-azure_connection::object_exists(const std::string &object_key, bool &exists) const
+azure_connection::object_exists(
+  const std::string &object_key, bool &exists, size_t &object_size) const
```

Comment:
I'm not a huge fan of having this function do two things. (If you've discussed this with Jie and considered the alternative, let me know - the last thing I want to do is stooge you around.)

The reason is that the callers currently seem to want either the existence, or the size - not both at the same time. For example, the call in `azure_file_open` needs a `WT_UNUSED` for the size. This also means the function can't throw an exception when only being used to check the size of an object, which means callers that want the size need to manually check `exists` and throw their own error.

Reply:
I discussed this with Jie and considered implementing `azure_connection::object_size` method in the connection class but it had almost identical implementation to `object_exists`. To reduce code duplication and for consistency with GCP and S3 implementation, I changed the parameters for `object_exists`.

Reply:
@korteland what are your thoughts? I am happy to change it to it's own function if it feels cleaner.

Reply:
Ack, I'm fine to leave it as-is then. It's not a huge deal, I'll re-review.